sszgen/unmarshal.go

```diff
@@ -28,15 +28,13 @@ func (v *Value) unmarshal(dst string) string {
 	case TypeContainer:
 		return v.umarshalContainer(false, dst)
 
-	case TypeBytes:
+	case TypeBitList, TypeBytes:
 		// both fixed and dynamic are decoded equally
-		return fmt.Sprintf("::.%s = append(::.%s, %s...)", v.name, v.name, dst)
+		return fmt.Sprintf("if cap(::.%s) == 0 {\n::.%s = make([]byte, 0, len(%s))\n}\n::.%s = append(::.%s, %s...)",  v.name, v.name, dst, v.name, v.name, dst)
```

Comment:
Instead of populating the generated code with "complex" logic, I usually prefer to add helper functions to the fastssz root.
Lets add a new function in encode.go that does this append:
```
// AppendBytes appends bytes from src to dst
func AppendBytes(dst, src []byte) []byte {
	if cap(dst) == 0 {
		dst = make([]byte, 0, len(src))
	}
	return append(dst, src...)
}
```
and then here in the generator we do:
```
return fmt.Sprintf("::.%s = ssz.AppendBytes(::.%s[:0], %s)", v.name, v.name, dst)
```

Reply:
I was having some issues with using that helper function. Does it matter that much if it is generated code? it would be inlined anyway.

Reply:
I like to keep the generated code as simple and short as possible because it helps with file size, debugging and aesthetics. What type of problem did you have with the helper function? I do not mind merging this PR and adding a new one later on with the helper function.